### DISCUSSION TITLE
Support for a custom OpenAI API URL

### DISCUSSION THREAD

**Discussion by homjay** [pieces-app/support#395] [Ideas]
- Hi, Pieces is a great work. Although it supports a custom OpenAI key, I still wish it could support a custom API URL. This would allow user to deploy the Pieces service with an OpenAI-compatible API.

**Comment by Erquint** [comment]
I was under the impression that Pieces can function as a middleman between an editor plugin and any inference endpoint compliant with OpenAI API. Yet found myself confusedly looking for the option after installation.  

I'd like to use Codestral remote API provided by Mistral AI in Sublime Text 4. Either the gratis API that Mistral AI provides for Codestral alone or the Le Platforme API that has Codestral as one of their provided models.  

But I'm seeing only hardcoded API endpoints — which I think are proxied through your servers — and local Ollama-hosted inference options. Nowhere to input the endpoint address and token for Mistral AI Codestral or any other inference endpoint and model I'd prefer to use.

That kind of capability would also serve for locally-hosted OAI-compliant inference servers other than Ollama.

**Comment by demianmnave** [comment]
This is a non-starter for me, unfortunately, as it would be for other users with data protection requirements (e.g. those operating in sovereign clouds).

I would be a paying user with this functionality.
